### DISCUSSION TITLE
(HampelFilter) Input doesn't equal output

### DISCUSSION THREAD

**Discussion by shadynash** [alan-turing-institute/sktime#2430] [Q&A]
my dataset: [Milling Data set.zip](https://github.com/alan-turing-institute/sktime/files/8459233/Milling.Data.set.zip)

```
m = sio.loadmat(r'D:\THESIS\MY\Milling Data set\mill without Nan , all eq 9000.mat',struct_as_record=True)
##Keys:  dict_keys(['__header__', '__version__', '__globals__', 'mill'])##
data = m['mill'] #shape is(1, 167)
```
![Capture](https://user-images.githubusercontent.com/57572781/162625554-b8be22db-aa33-458a-9eb6-f043d9436f7f.PNG)
this picture shows the original dataset from MATLAB maybe easier to understand the dataset (i am using Python)

```
import sktime
from sktime.transformations.series.outlier_detection import HampelFilter
from sktime.datatypes import check_raise  
```

```
X=data[0,:]['vib_spindle']    # I chose only the column for **vib_spindle**

def Hampel(H):
    return HampelFilter(window_length=10).fit_transform(H)
data_vib_spindle= X

filterd_vib_spindle=[ ]
for H in data_vib_spindle:
    a=Hampel(H)
    filterd_vib_spindle.append(a)
```
the **X** was a ndarray (1, 167) and the output  **filterd_vib_spindle** is list    (167)
now the problem is when i convert the list  **filterd_vib_spindle** to array because i want to add it to the dataset it became a flat array 

`np.asarray(filterd_vib_spindle)`
now the shape is (167, 9000, 1)

**How can i return the shape of an array after converting as before??** 
i wish you can help me to pointed my mistake or guide me to the right path. your help and time really appreciated

**Comment by fkiraly** [comment]
@shadynash, really sorry, I think I'm getting a bit confused here - I feel a reason is that I can't run your lines without your data.

Would you kindly be able to:
* create some dummy code that is self-contained and has the same problematic behaviour (so not reliant on your data but creating a data frame or similar from scratch)
* paste it here, in a code block (that I can copy-paste from to run)
* explain what you *expect* the output to be
?
